mesonbuild/compilers/cs.py

```diff
@@ -40,9 +40,8 @@ class CsCompiler(BasicLinkerIsCompilerMixin, Compiler):
     language = 'cs'
 
     def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoice,
-                 info: 'MachineInfo', comp_id: str, runner: T.Optional[str] = None):
+                 info: 'MachineInfo', runner: T.Optional[str] = None):
         super().__init__(exelist, version, for_machine, info)
-        self.id = comp_id
```

Comment:
Lol, why were we passing the subclass id around as a parameter to `__init__`. Even worse than the garden variety use of `self.id = '...'` in extraneous init functions. :D